<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>

  <!-- Node Settings -->
  <arg name="debug" default="false"/>
  <arg name="output_frame" default="local_origin"/>
  <arg name="use_velocity" default="false"/>
  <arg name="output_framerate" default="10"/>

  <arg name="indoor" default="false"/>
  <arg name="use_two_cameras" default="true">

  <group ns="$(arg uav_name)">
  
    <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
      <node pkg="uvdar" type="uvdar_kalman_anonymous_node" name="filterUvdar" output="screen">
        <param name="debug" value="$(arg debug)" />
        <param name="uav_name" value="$(arg uav_name)" />
        <param name="output_frame" value="$(arg output_frame)"/>
        <param name="output_framerate" value="$(arg output_framerate)" />

        <param name="indoor" value="$(arg indoor)" />
        <param name="use_velocity" value="$(arg use_velocity)" />

        <group if="$(arg use_two_cameras)">
          <rosparam param="measured_poses_topics"> ["measured_poses_L", "measured_poses_R"] </rosparam>
          <remap from="~measured_poses_L" to="/$(arg uav_name)/uvdar/measuredPosesL"/>
          <remap from="~measured_poses_R" to="/$(arg uav_name)/uvdar/measuredPosesR"/>
        </group>
        <group unless="$(arg use_two_cameras)">
          <rosparam param="measured_poses_topics"> ["measured_poses"] </rosparam>
          <remap from="~measured_poses" to="/$(arg uav_name)/uvdar/measuredPoses"/>
        </group>

        <remap from="~filtered_poses" to="/$(arg uav_name)/uvdar/filteredPoses"/>

        <remap from="~filtered_poses/tentative" to="/$(arg uav_name)/uvdar/filteredPoses/tentative"/>

      </node>
    </group>

  </launch>
